internal/telegram/chatbot.go

```diff
@@ -1,34 +1,74 @@
 package telegram
 
-import tgbotapi "github.com/go-telegram-bot-api/telegram-bot-api/v5"
+import (
+	"backend/internal/configs"
+	"backend/internal/handler/tgmessagehandler"
+	"backend/internal/ws"
 
-// TODO: Add more functionality to this echo only bot
-func StartChatbot(token string) {
-	bot, err := tgbotapi.NewBotAPI(token)
+	tgbotapi "github.com/go-telegram-bot-api/telegram-bot-api/v5"
+)
+
+var botAPI *tgbotapi.BotAPI
+
+func GetBotAPI() *tgbotapi.BotAPI {
+	return botAPI
+}
+
+// Documentation: https://pkg.go.dev/github.com/go-telegram-bot-api/telegram-bot-api/v5#section-documentation
+// Getting started: https://go-telegram-bot-api.dev/
+
+func StartChatbot(cfg *configs.Config, hub *ws.Hub) {
+	bot, err := tgbotapi.NewBotAPI(cfg.TelegramToken)
 	if err != nil {
 		panic(err)
+	} else {
+		botAPI = bot
 	}
 
-	// Uncomment this line when debugging the telegram bot
-	// bot.Debug = true
+	if cfg.GoEnv == "development" {
+		bot.Debug = true
+	}
 
 	updateConfig := tgbotapi.NewUpdate(0)
-
 	updateConfig.Timeout = 30
 
 	updates := bot.GetUpdatesChan(updateConfig)
+	handleUpdates(bot, hub, updates)
+}
```

Comment:
```suggestion
	updateConfig := tgbotapi.NewUpdate(0)
	updateConfig.Timeout = 30

	setCommands(bot)

	updates := bot.GetUpdatesChan(updateConfig)
	handleUpdates(bot, hub, updates)
}

func setCommands(bot *tgbotapi.BotAPI) {
	commandsConfig := tgbotapi.NewSetMyCommandsWithScopeAndLanguage(
		tgbotapi.NewBotCommandScopeAllPrivateChats(),
		"en",
		tgbotapi.BotCommand{Command: tgmessagehandler.HelpCmdWord, Description: tgmessagehandler.HelpCmdDesc},
		tgbotapi.BotCommand{Command: tgmessagehandler.AuthCmdWord, Description: tgmessagehandler.AuthCmdDesc},
		tgbotapi.BotCommand{Command: tgmessagehandler.QueryCmdWord, Description: tgmessagehandler.QueryCmdDesc},
		tgbotapi.BotCommand{Command: tgmessagehandler.RequestCmdWord, Description: tgmessagehandler.RequestCmdDesc},
	)

	msg, err := bot.Request(commandsConfig)
	log.Info(msg)
	if err != nil {
		log.Fatalf("error setting commands: %s", err.Error())
	}
}
```

<img width="664" alt="image" src="https://github.com/DYOC-Chatbot/chatbot-backend/assets/34371483/50bd8f62-645b-44ec-8bd1-b0496d6b855b">

(small addition - add command config)

for future reference: i just figured out that all configs we can send are a `Chattable` and we can use bot.Request to set them :^D

did not include /start

Reply:
That a great addition